.github/workflows/build-and-deploy.yml

```diff
@@ -78,19 +69,19 @@ jobs:
       - name: Determine user-service rebuild status
         id: user-service
         run: |
-          echo 'rebuild=${{ needs.detect-code-changes.outputs.user-service == 'true' }}' >> $GITHUB_OUTPUT
+          echo 'rebuild=${{ needs.detect-code-changes.outputs.user-service == 'true' || needs.detect-code-changes.outputs.ci == 'true' }}' >> $GITHUB_OUTPUT
```

Comment:
We now rebuild the services also when the CI changes. This is not 100% needed for this PR but it seems like a good idea as most likely if the CI changes there's a chance that this impacts how the docker images are build for the services.